src/loaders/common/callback_server_wrapper.py

```diff
@@ -17,69 +17,89 @@ class Conf:
+        token_filepath: str | None = None,
+        au_service_ver: str = "release",
+        workers: int = 5,
+        max_task: int = 20,
+        worker_function: Callable | None = None,
+        retrieve_sample: bool = False,
+        ignore_no_sample_error: bool = False,
+        workspace_downloader: bool = True,
     ) -> None:
         """
         Initialize the configuration class.
 
         Args:
             job_dir (str): The directory for SDK jobs per user.
             output_dir (str): The directory for a specific workspace id under sourcedata/ws.
-            worker_function (Callable): The function that will be called by the workers.
             kb_base_url (str): The base url of the KBase services.
-            token_filepath (str): The file path that stores a KBase token appropriate for the KBase environment. 
+            token_filepath (str): The file path that stores a KBase token appropriate for the KBase environment.
                                 If not supplied, the token must be provided in the environment variable KB_AUTH_TOKEN.
+            au_service_ver (str): The service verison of AssemblyUtilClient.
             workers (int): The number of workers to use for multiprocessing.
+            max_task (int): The maxmium subtasks for the callback server.
+            worker_function (Callable): The function that will be called by the workers.
             retrieve_sample (bool): Whether to retrieve sample for each genome object.
             ignore_no_sample_error (bool): Whether to ignore the error when no sample data is found.
+            workspace_downloader (bool): Whether to be used for the workspace downloader script.
         """
+        # common instance variables
+        ipv4 = loader_helper.get_ip()
         port = loader_helper.find_free_port()
         token = loader_helper.get_token(token_filepath)
-        self.retrieve_sample = retrieve_sample
-        self.ignore_no_sample_error = ignore_no_sample_error
-        ipv4 = loader_helper.get_ip()
+
+        ws_url = os.path.join(kb_base_url, "ws")
+        callback_url = "http://" + ipv4 + ":" + str(port)
+        print("callback_url:", callback_url)
+
         self._start_callback_server(
             docker.from_env(),
             uuid.uuid4().hex,
             job_dir,
             kb_base_url,
             token,
             port,
+            max_task,
             ipv4,
         )
 
-        ws_url = os.path.join(kb_base_url, "ws")
-        sample_url = os.path.join(kb_base_url, "sampleservice")
-        callback_url = "http://" + ipv4 + ":" + str(port)
-        print("callback_url:", callback_url)
-
         self.ws = Workspace(ws_url, token=token)
-        self.asu = AssemblyUtil(callback_url, token=token)
-        self.ss = SampleService(sample_url, token=token)
+        self.asu = AssemblyUtil(callback_url, service_ver=au_service_ver, token=token)
 
-        self.workers = workers
```

Comment:
No longer needed?

Reply:
workers are only required for the downloader

Reply:
But the downloader needs this file, right? so workers still needs to be there

Reply:
https://github.com/kbase/collections/blob/f515968245e44ba7cbe1ee012acb1d62e3697dbd/src/loaders/common/callback_server_wrapper.py#L93

Reply:
so workers doesn't need to be an instance variable?

Reply:
no